packages/headless/rollup.config.js

```diff
@@ -114,38 +111,70 @@ const browser = [
     input: 'src/index.ts',
     output: [
       buildUmdOutput('dist/browser', 'CoveoHeadless'),
-      buildEsmOutput('dist/browser')
-    ]
+      buildEsmOutput('dist/browser'),
+    ],
   },
   {
     input: 'src/case-assist.index.ts',
     output: [
       buildUmdOutput('dist/browser/case-assist', 'CoveoHeadlessCaseAssist'),
-      buildEsmOutput('dist/browser/case-assist')
-    ]
+      buildEsmOutput('dist/browser/case-assist'),
+    ],
   },
   {
     input: 'src/recommendation.index.ts',
     output: [
-      buildUmdOutput('dist/browser/recommendation', 'CoveoHeadlessRecommendation'),
-      buildEsmOutput('dist/browser/recommendation')
-    ]
+      buildUmdOutput(
+        'dist/browser/recommendation',
+        'CoveoHeadlessRecommendation'
+      ),
+      buildEsmOutput('dist/browser/recommendation'),
+    ],
   },
   {
     input: 'src/product-recommendation.index.ts',
     output: [
-      buildUmdOutput('dist/browser/product-recommendation', 'CoveoHeadlessProductRecommendation'),
-      buildEsmOutput('dist/browser/product-recommendation')
-    ]
+      buildUmdOutput(
+        'dist/browser/product-recommendation',
+        'CoveoHeadlessProductRecommendation'
+      ),
+      buildEsmOutput('dist/browser/product-recommendation'),
+    ],
   },
   {
     input: 'src/product-listing.index.ts',
     output: [
-      buildUmdOutput('dist/browser/product-listing', 'CoveoHeadlessProductListing'),
-      buildEsmOutput('dist/browser/product-listing')
-    ]
+      buildUmdOutput(
+        'dist/browser/product-listing',
+        'CoveoHeadlessProductListing'
+      ),
+      buildEsmOutput('dist/browser/product-listing'),
+    ],
   },
-].filter(b => matchesFilter(b.input)).map(buildBrowserConfiguration);
+]
+  .filter((b) => matchesFilter(b.input))
+  .map(buildBrowserConfiguration);
+
+const e2e = [
+  {
+    input: 'src/index.ts',
+    output: [
+      buildEsmOutput('dist/browser'),
+      buildUmdOutput('dist/browser', 'CoveoHeadless'),
+    ],
+  },
+]
+  .filter((b) => matchesFilter(b.input))
+  .map(buildBrowserConfiguration);
+
+const e2eNode = [
+  {
+    input: 'src/index.ts',
+    outDir: 'dist',
+  },
+]
+  .filter((b) => matchesFilter(b.input))
+  .map(buildNodeConfiguration);
```

Comment:
Building just the `search` is okay for now, but if Atomic one day wants to support e.g. recommendation, then we would need to add that use-case to able to e2e test it? At the limit, the `e2e` and `prod` build times would become the same?


Sticking with the current approach, what if we filter for just the `search` use-case instead of duplicating the config? 

````
const e2eNode = nodejs.filter(config => isSearchUseCase)
const e2eBrower =  ...
````

I'd be interested to know how fast webpack could create the bundles too. If it can minify quickly, that could speed up the build and simplify the config.

I think if this PR focused on just the cypress part, then we explored speeding up the headless build in more depth in a separate PR, perhaps we might come to a different conclusion 🤷